thread/common/omrthread.c

```diff
@@ -4360,7 +4360,8 @@ monitor_exit(omrthread_t self, omrthread_monitor_t monitor)
 #if defined(OMR_THR_SPIN_WAKE_CONTROL)
 		omrthread_spinlock_swapState(monitor, J9THREAD_MONITOR_SPINLOCK_UNOWNED);
  		MONITOR_LOCK(monitor, CALLER_MONITOR_EXIT1);
- 		if (0 == monitor->spinThreads) {
+		issueReadWriteBarrier();
+		if (0 == monitor->spinThreads) {
```

Comment:
I don't expect the barrier to help. Any threads that have determined that they should spin but have not yet incremented `spinThreads` will continue to have the problem this aims to correct.